testsuite/Properties.hs

```diff
@@ -0,0 +1,30 @@
+module Main (
+    main
+  ) where
+
+import Test.Tasty
+import Test.Tasty.QuickCheck as QC
+
+import Morte.Core
+import ClosedWellTyped
+
+main :: IO ()
+main = defaultMain tests
+
+tests :: TestTree
+tests = testGroup "Properties" [
+      QC.testProperty "normalize . normalize == normalize" normalizeIdempotent 
+    , QC.testProperty "Well typed after normalization"     wellTypedAfterNormalize
+    ]
+
+wellTypedAfterNormalize :: ClosedWellTyped -> Bool
+wellTypedAfterNormalize (ClosedWellTyped expr) = 
+        not (isRight $ typeOf expr) || (isRight $ typeOf $ normalize expr)
```

Comment:
Instead of `not x || y`, use `x ==> y`.  `QuickCheck` will then ensure that `x` is true a high enough fraction of time to exercise the `y` expression